syntax/scan.go

```diff
@@ -861,15 +864,26 @@ func (sc *scanner) scanNumber(val *tokenValue, c rune) Token {
 		return FLOAT
 	} else {
 		var err error
+		var ok bool = true
```

Comment:
Move this line down to just before where it is needed.  You don't need to set it to true since you'll immediately reassign it.